packages/ui/src/views/entity-reference/entity-reference-component.tsx

```diff
@@ -0,0 +1,217 @@
+        onClick={() => onSelect?.(parentFolder)}
+        thumbnail={<Icon name="circle-arrow-top" size={16} className="text-foreground-5" />}
+      >
+        <StackedListField title={<span className="capitalize">{String(parentFolder)}</span>} />
+      </StackedListItem>
+    )
+  }
+
+  // Default folder renderer
+  const defaultFolderRenderer = ({ folder, onSelect }: FolderRendererProps<F>) => {
+    return (
+      <StackedListItem
+        onClick={() => onSelect?.(folder)}
+        thumbnail={<Icon name="folder" size={16} className="text-foreground-5" />}
+      >
+        <StackedListField title={<span className="capitalize">{String(folder)}</span>} />
+      </StackedListItem>
+    )
+  }
+
+  const renderCombinedList = () => {
+    return (
+      <StackedList>
+        {/* scopes */}
+        {parentFolder && (
+          <>
+            {renderScopeSelector
+              ? renderScopeSelector({
+                  parentFolder,
+                  onSelect: () => handleScopeChange(DirectoryType.UP)
+                })
+              : defaultScopeSelectorRenderer({
+                  parentFolder,
+                  onSelect: () => handleScopeChange(DirectoryType.UP)
+                })}
+          </>
+        )}
+
+        {/* folders */}
+        {childFolder && (
+          <>
+            {renderFolder
+              ? renderFolder({
+                  folder: childFolder,
+                  onSelect: () => handleScopeChange(DirectoryType.DOWN)
+                })
+              : defaultFolderRenderer({
+                  folder: childFolder,
+                  onSelect: () => handleScopeChange(DirectoryType.DOWN)
+                })}
+          </>
+        )}
+
+        {/* entities */}
+        {entities.length > 0 ? (
+          <>
+            {entities.map(entity => {
+              const isSelected = entity.id === selectedEntity?.id
+
+              return (
+                <React.Fragment key={entity.id}>
+                  {renderEntity
+                    ? renderEntity({
+                        entity,
+                        isSelected,
+                        onSelect: handleSelectEntity
+                      })
+                    : defaultEntityRenderer({
+                        entity,
+                        isSelected,
+                        onSelect: handleSelectEntity
+                      })}
+                </React.Fragment>
+              )
+            })}
+          </>
+        ) : (
+          <StackedListItem disableHover>
+            <StackedListField
+              title={<div className="text-foreground-4 flex h-32 items-center justify-center">No items available</div>}
+            />
+          </StackedListItem>
+        )}
+      </StackedList>
+    )
+  }
+
+  return (
+    <>
+      {isLoading ? (
+        <SkeletonList />
+      ) : (
+        <>
+          {showFilter && <Input type="text" placeholder="Search" className="mb-4" />}
+          <ScrollArea className="h-[62%]">
```

Comment:
might be better to avoid a fixed relative height. and just let it expand to fill the parent